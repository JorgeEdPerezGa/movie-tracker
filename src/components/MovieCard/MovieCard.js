import React from 'react';

const MovieCard = ({ title, overview, poster, backdrop }) => {
  return (
    <article className='movie-card'>
      <p className='movie-title'>{ title }</p>
      <p className='movie-overview'>{ overview }</p>
      <img
        className='movie-poster' src={`https://image.tmdb.org/t/p/w500${poster}`}/>
      <img
        className='movie-backdrop' src={`https://image.tmdb.org/t/p/w500${backdrop}`}/>

    </article>
  );
};

export default MovieCard;
