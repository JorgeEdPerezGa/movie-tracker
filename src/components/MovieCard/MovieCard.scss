.movie-card-main {
  width: 300px;
  background: #393839;
  margin: 15px;
  height: 510px;
}
.movie-poster {
  width: 300px;
}

.favorite-button {
  background: url('../../assets/favorite.svg');
  border: none;
  width: 35px;
  height: 35px;
  margin-top: 12px;
    &:hover {
        background: url('../../assets/favorite-selected.svg');
        cursor: pointer;
    }
}
