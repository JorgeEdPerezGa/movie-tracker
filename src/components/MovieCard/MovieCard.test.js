import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MovieCard', () => {
  it.skip('should not pass', () => {
    expect(false).toEqual(true);
  })
})