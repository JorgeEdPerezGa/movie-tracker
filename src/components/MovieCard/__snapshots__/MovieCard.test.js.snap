// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieCard
    movie={
        Object {
            "backdrop_path": "/askg3SMvhqEl4OL52YuvdtY40Yb.jpg",
            "movie_id": 354912,
            "overview": "Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz.",
            "poster_path": "/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
            "release_date": "2017-10-27",
            "title": "Coco",
            "vote_average": 7.7,
          }
    }
    onFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/movies/354912"
>
          <img
                    className="movie-poster"
                    src="https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
          />
</Link>,
        <div
          className="button-container"
>
          <button
                    className="favorite-button "
                    onClick={[Function]}
          />
</div>,
      ],
      "className": "movie-card-main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            className="movie-poster"
            src="https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
/>,
          "replace": false,
          "to": "/movies/354912",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "movie-poster",
            "src": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="favorite-button "
            onClick={[Function]}
/>,
          "className": "button-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorite-button ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/movies/354912"
>
            <img
                        className="movie-poster"
                        src="https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
            />
</Link>,
          <div
            className="button-container"
>
            <button
                        className="favorite-button "
                        onClick={[Function]}
            />
</div>,
        ],
        "className": "movie-card-main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              className="movie-poster"
              src="https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg"
/>,
            "replace": false,
            "to": "/movies/354912",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "movie-poster",
              "src": "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="favorite-button "
              onClick={[Function]}
/>,
            "className": "button-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "favorite-button ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
